ui/tests/e2e/login.test.ts

```diff
@@ -0,0 +1,56 @@
+import puppeteer from 'puppeteer'
+import { LOGIN_URL, OVERVIEW_URL, PUPPETEER_CONFIG } from './test_config'
+
+describe('Login', () => {
+  let browser
+  beforeAll(async () => {
+    browser = await puppeteer.launch(PUPPETEER_CONFIG)
+  })
+
+  afterAll(() => {
+    browser.close()
+  })
+
+  it(
+    'should login fail by incorrect password',
+    async () => {
+      const page = await browser.newPage()
+      await page.goto(LOGIN_URL)
+
+      const pwdInput = await page.waitForSelector('input#tidb_signin_password')
+      await pwdInput.type('any')
+      await page.click('button#signin_btn')
+
+      const errorContainer = await page.waitForSelector('div.has-error')
+      const containsPwdInput = await errorContainer.$eval(
+        'input#tidb_signin_password',
+        (el) => (el ? true : false)
+      )
+      expect(containsPwdInput).toBe(true)
+      const errorContent = await errorContainer.$eval(
+        '.ant-form-explain',
```

Comment:
please avoid introducing ant design specific selectors, which does not have compatibility guarantees.